### PR TITLE
Adding partitioning support for OTA

### PR DESCRIPTION
**Pull request by engdavidiogo**
## Summary

Added structure for partition initialization. 
Purpose of integration with MCUboot and ESP32C6.

## Impact

RISC-V/ESP32

## Testing

 ./tools/configure.sh esp32c6-devkitc:mcuboot_nsh

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)

__Fill In The Commit Message:__ This PR contains a Commit with an Empty Commit Message. Please fill in the Commit Message with the PR Summary.



**No, this PR does not meet the NuttX Requirements.** 

Here's why and how to improve it:

* **Insufficient Summary:**
    *  **Why is this change necessary?**  Clearly state if this fixes a bug, adds a new feature (MCUboot and ESP32C6 support), or is an update.
    * **What functionality changes?**  Be specific about the code being modified (e.g., partition management, bootloader integration).
    * **How does it work?** Provide a technical explanation of the structure and its role in initialization. 
    * **Link related issues:** If this relates to existing issues on the NuttX or NuttX Apps repositories, include the issue numbers.

* **Incomplete Impact Assessment:**
    * **New/Changed Features:**  Explicitly state that this adds MCUboot and ESP32C6 support. 
    * **User Impact:** Will users need to change configurations, commands, or workflows?
    * **Build Impact:** Are there new build steps, dependencies, or configuration options?
    * **Hardware Impact:** Specify the affected architectures (RISC-V) and boards (ESP32C6).
    * **Documentation Impact:** Does this require documentation updates? If so, confirm if they are included in the PR.
    * **Security, Compatibility:** Address these aspects even if there's no impact. State "NO" explicitly for clarity.

* **Insufficient Testing Information:**
    * **Build Host Details:** Provide specifics about your development environment (OS, CPU architecture, compiler version).
    * **Target Details:** Include the full target configuration (e.g., `esp32c6-devkitc:mcuboot_nsh`)
    * **Meaningful Test Logs:** The provided logs don't demonstrate the functionality of the partition initialization. Include logs that show:
        * Partition creation/mounting (before and after the change)
        * Interaction with MCUboot (if applicable)
        * Any relevant output that verifies the structure is working as intended. 

**Recommendations:**

1. **Expand the Summary:**  Provide a concise but thorough overview of the change's purpose, functionality, and how it works.
2. **Complete the Impact Assessment:** Address all impact categories with specific details.
3. **Provide Detailed Testing Information:** Include comprehensive information about your build and target environments. 
4. **Show Verifying Test Logs:** Demonstrate the functionality of your code changes with clear and relevant logs.

**Comment by tmedicci** [comment]
I'm on vacation until next Wednesday (Oct, 16th), but I'm unsure if we need to create the partitions for the current firmware. If I'm not mistaken, MCUboot can handle it directly.

@almir-okato, @fdcavalcanti and @eren-terzioglu , can you PTAL?

(@acassis , tag them for NuttX-related issues while I'm off, please).

**Comment by engdavidiogo** [comment]
Hi everyone,

Please let me know if I missed anything. When building the binary with the latest version of **MCUboot** to **ESP32C6**, it was unable to handle the partitioning correctly, which is why the proposed improvement was made.

Once again, feel free to let me know what else needs to be done. 
I’ll be happy to resolve any issues.

File: boards/risc-v/esp32c6/common/src/esp_board_spiflash.c [hunk 1]
```diff
@@ -58,18 +58,102 @@
...
+static int init_ota_partitions(void)
+{
+  struct mtd_dev_s *mtd;
+  int ret = OK;
+
+#ifdef CONFIG_BCH
+  char blockdev[18];
+#endif
+
+  for (int i = 0; i < nitems(g_ota_partition_table); ++i)
```
Comment (xiaoxiang781216) [inline boards/risc-v/esp32c6/common/src/esp_board_spiflash.c:129]: move int i to the beginning of the function

Comment (engdavidiogo) [reply boards/risc-v/esp32c6/common/src/esp_board_spiflash.c]: adjusted

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by almir-okato** [comment]
Hi @engdavidiogo 
Please run the `checkpatch` tool from NuttX locally and fix the style issues pointed by that, also sign your commits (use `-s` param when commiting or when amending). These may fix the failing "Check" step from CI.

**Comment by almir-okato** [comment]
@engdavidiogo last suggestions, please squash your commits to one, put its commit message like "boards: riscv: esp32c6: <title of the commit>" and the commit description. 
Its for keeping the commit pattern, you can check the git history from the repo as example.

**Comment by engdavidiogo** [comment]
Hi @almir-okato , thank you for your guidance.
I am observing the community and trying to absorb the processes.
I am available for further improvements.

**Comment by xiaoxiang781216** [comment]
squash to the change into one patch.

**Comment by tmedicci** [comment]
Hi @engdavidiogo , I was able to test it and validate it works for OTA on ESP32-C6. Can you please squash your commits and rebase  to test it on CI and merge?

**Review by tmedicci — APPROVED** [review]

**Comment by acassis** [comment]
@tmedicci it is better to keep two separated commits, because the second will be useful for people see how to add it to new boards.

**Pull request merged**